src/vs/workbench/contrib/debug/browser/debugService.ts

```diff
@@ -279,11 +279,17 @@ export class DebugService implements IDebugService {
 			// make sure to save all files and that the configuration is up to date
 			await this.extensionService.activateByEvent('onDebug');
 			if (!options?.parentSession) {
-				await this.editorService.saveAll();
-				const activeEditor = this.editorService.activeEditorPane;
-				if (activeEditor) {
-					// Make sure to save the active editor in case it is in untitled file it wont be saved as part of saveAll #111850
-					await this.editorService.save({ editor: activeEditor.input, groupId: activeEditor.group.id });
+				const saveBeforeStartConfig = this.configurationService.getValue('debug.saveBeforeStart');
+
+				if (saveBeforeStartConfig !== 'none') {
+					await this.editorService.saveAll();
+					if (saveBeforeStartConfig === 'allEditorsInActiveGroup') {
+						const activeEditor = this.editorService.activeEditorPane;
+						if (activeEditor) {
```

Comment:
Please restructure this a bit such that you do not save all editors twice.

For example 
```
if (saveBeforeStartConfig === 'nonUntitledEditorsInActiveGroup') {
  .saveAll()
}
if (saveBeforeStartConfig === 'allEditorsInActiveGroup') {
  activeEditor...
}
```

Reply:
I don't see how the second branch in your comment is supposed to work. Judging by the previous code, `saveAll` is still needed to save everything in both cases, and the `allEditorsInActiveGroup` case just adds another save to the current one.

This is the logic you wrote in cf8ed37206e161a30151defa4fc9ea58c32335bc, where did I introduce a double save?

Reply:
You are correct, I was mistaken. Please ignore my comment.